lib/davinci_crd_test_kit/client_tests/submitted_response_validation.rb

```diff
@@ -0,0 +1,42 @@
+module DaVinciCRDTestKit
+  class SubmittedResponseValidationTest < Inferno::Test
+    include CardsValidation
+
+    title 'Custom CDS Service Response is valid'
+    id :crd_submitted_response_validation
+
+    input :custom_response, optional: true
+
+    def hook_name
+      config.options[:hook_name]
+    end
+
+    def response_label(_index = nil)
+      'Custom response'
+    end
+
+    def valid_cards
+      @valid_cards ||= []
+    end
+
+    def validate_system_actions(system_actions)
+      return if system_actions.blank?
+
+      system_actions.each do |action|
+        action_fields_validation(action)
+      end
+    end
+
+    run do
+      omit_if custom_response.blank?, 'Custom response was not provided'
+
+      assert_valid_json custom_response
+
+      custom_response_hash = JSON.parse(custom_response)
+
+      perform_cards_validation(custom_response_hash['cards'])
+
+      validate_system_actions(custom_response_hash['systemActions'])
```

Comment:
One thing I am noticing is that this submitted response validation does not do any validation based on the type of hook the cards response is for. This may be confusing if someone thinks that the card response they have inputted is correct, but it is actually missing important requirements according to the hook type it is supposed to be a card response for.

Not sure how in depth we want this submitted response validation to be, but when submitting a cards response as an input, you can get it to pass this test, but in the Inferno CRD server tests there will be failures due to certain cards not being included that are required for that hook type. For example, the Appointment Book hook requires an Instructions card to be returned, but this test will not fail if an Instructions card is not included in the submitted Appointment Book card response input. I think it would make sense to do some additional validation here that checks if all the required cards are contained for the specific hook type that the cards response is for.

Reply:
I think we'll want a separate ticket to figure out exactly what sort of validation we should do on the submitted responses.

Reply:
Alright, then in that case everything is looking good. I can re-review once unit tests are done

Reply:
I added some basic unit tests. The majority of the checking behavior is already tested by `service_response_validation_test_spec`.